src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingNamedTypeSymbol.cs

```diff
@@ -250,34 +250,31 @@ private static ExtendedErrorTypeSymbol CyclicInheritanceError(RetargetingNamedTy
             return new ExtendedErrorTypeSymbol(declaredBase, LookupResultKind.NotReferencable, info, true);
         }
 
-        internal override NamedTypeSymbol BaseTypeNoUseSiteDiagnostics
+        internal override NamedTypeSymbol GetBaseTypeNoUseSiteDiagnostics(bool ignoreNonNullTypesAttribute)
         {
-            get
+            if (ReferenceEquals(_lazyBaseType, ErrorTypeSymbol.UnknownResultType))
             {
-                if (ReferenceEquals(_lazyBaseType, ErrorTypeSymbol.UnknownResultType))
-                {
-                    NamedTypeSymbol acyclicBase = GetDeclaredBaseType(null);
-
-                    if ((object)acyclicBase == null)
-                    {
-                        // if base was not declared, get it from BaseType that should set it to some default
-                        var underlyingBase = _underlyingType.BaseTypeNoUseSiteDiagnostics;
-                        if ((object)underlyingBase != null)
-                        {
-                            acyclicBase = this.RetargetingTranslator.Retarget(underlyingBase, RetargetOptions.RetargetPrimitiveTypesByName);
-                        }
-                    }
+                NamedTypeSymbol acyclicBase = GetDeclaredBaseType(null);
```

Comment:
>GetDeclaredBaseType(null); [](start = 46, length = 26)

`GetDeclaredBaseType(null, ignoreNonNullTypesAttribute);`

Reply:
From our discussion: I think it would make sense to pass `ignoreNonNullTypesAttribute`, but it's unobservable to my knowledge. I only passed it through where it solved a problem so far.

---
In reply to: [201402784](https://github.com/dotnet/roslyn/pull/28320#discussion_r201402784) [](ancestors = 201402784)